code/game/objects/items/weapons/RCD.dm

```diff
@@ -398,27 +401,26 @@ RCD
 	color = "red"
 
 
+
 // Ranged RCD
 
 
+
 /obj/item/weapon/rcd/arcd/afterattack(atom/A, mob/user, proximity)
-	proximity = 1
 	if(!(A in view(7, get_turf(user))))
 		to_chat(user, "<span class='warning'>The \'Out of Range\' light on the RCD blinks red.</span>")
 		return 0
-	if(isturf(A) || istype(A, /obj/machinery/door/airlock) || istype(A, /obj/structure/grille) || istype(A, /obj/structure/window) || istype(A, /obj/structure/girder))
+	if((isturf(A) && mode==3) || (isturf(A) && mode==1 && !A.density) || (istype(A, /obj/machinery/door/airlock) && mode==3) || istype(A, /obj/structure/grille) || (istype(A, /obj/structure/window) && mode==3) || istype(A, /obj/structure/girder))
```

Comment:
Thats means I have to add defines to the RCD too, the thing that Goof just refactored.